client/ListingDetails.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { Component } from 'react';
+import { shape, number, string, array, bool, func } from 'prop-types';
+import Modal from 'react-modal';
+import ReactStars from 'react-stars';
+import style from './listingdetails.css';
+
+class ListingDetails extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  componentWillMount() {
```

Comment:
You should touch base with Danny as to why this lifecycle method is considered unsafe.

https://hackernoon.com/problematic-react-lifecycle-methods-are-going-away-in-react-17-4216acc7d58b

Reply:
I think this is a specific issue with the modal library I chose:
https://github.com/reactjs/react-modal/issues/133
It was the only fix that worked because of how the modal attaches to the DOM but I'll revisit this later. Thanks for the advice!